packages/zoe/src/contracts/autoswap.js

```diff
@@ -122,14 +122,14 @@ const makeContract = zcf => {
         const liquidityAmountOut = liquidityAmountMath.make(liquidityExtentOut);
         const liquidityPaymentP = liquidityMint.mintPayment(liquidityAmountOut);
 
+        liqTokenSupply += liquidityExtentOut;
```

Comment:
Can you clarify why this being out here is the right thing?

Reply:
Actually, in writing this out, I convinced myself that it is not the right thing. So thanks for asking :) Here's my reasoning:

The `liqTokenSupply` is used to calculate percent ownership of the pool, given a certain number of liquidity tokens. I had been thinking that the contract should err on the side of adding to `liqTokenSupply` first, even if there are later problems. This was coming from the transfer pattern of always doing the subtraction from one account before adding to another account, for safety. But, if a user can artificially inflate the liqTokenSupply number, that has the effect of devaluing all other users' liquidity tokens, so this is wrong, and adding to the liqTokenSupply should be done after we confirm the other steps succeed.